server/endpoints/file/fetch.py

```diff
@@ -59,3 +70,26 @@ def action(project: Project, props: Props) -> Result:
         raise Exception("The file is not tabular")
 
     return Result(path=path)
+
+
+# We export Google Sheets to a temporary CSV file and read it
+def read_google_sheets_bytes(url: str):
+    table = TableResource(path=url)
+    with tempfile.NamedTemporaryFile(suffix=".csv") as file:
+        table.write(file.name)
+        resource = FileResource(path=file.name)
+        bytes = resource.read_file()
+        return bytes
+
+
+# We use public HTML to extract the title of the document
+def read_google_sheets_path(url: str):
+    file = FileResource(path=url)
+    text = file.read_text(size=10000)
+    match = re.search(r"<title>(.*?)</title>", text)
+    path = "google-sheets.csv"
+    if match:
+        title = match.group(1).strip()
+        title = title.replace(" - Google Sheets", "")
```

Comment:
@roll Amazing work!!! :100: also i like that you found another way of extracting the title that looks less hacky than what i suggested. 

One thing though. My locale is automatically set to Bulgarian and i get this title "Ethical Tech Job Resources - Google Таблици.csv" so i guess maybe it's better to replace anything after "- Google "? seems like the word "Sheets" is translated to the users language. What if the title itself includes this string? can we target the last ocurrence of this string instead, in the rare case that this happens more than once in the title?

Reply:
@guergana 
Great catch! I didn't know that they translate "Google Sheets". Can you please test it now with your locale?

We can also use `<meta property="og:title" content="table">` but it would probably require a proper HTML parsing library like BS to handle quotes escaping etc white title parsing is more straightforward.